### ISSUE TITLE
training with --random_weights generates unproportionally huge model file

### ISSUE THREAD

**Issue by matanster** [VowpalWabbit/vowpal_wabbit#1570]
With 8.6.1's release build, I get abnormal results training an otherwise normally training `cb_explore` model: 

Setting `random_weights` to `on`, once VW is done with going over the training dataset, the resulting model file is many orders of magnitude larger than without using `random_weights`: several **GB** v.s. only **few dozens of KB** when not training with `random_weights` set to `on`. 

My full training command is this one:
```
vw --data output/training_dataset.vw --cb_type dr --bit_precision 30 -f output/cb.model --random_weights on --cb_explore 12
```

I should provide a self-contained example, as my training data is as usual proprietary. 

I am _not_ getting this odd behavior when not using `random_weights` over the same dataset and training command. 

In case you're wondering, I'm using 30 bits for hashing as I get a few hash collisions even though the number of features I have is not huge (bad luck or related to https://github.com/JohnLangford/vowpal_wabbit/issues/1566? I can't tell yet), and I don't care about hashing speed as my training data is relatively small and training with my data is very fast (except for when using `random_weights` of course, as it takes a lot of time to build that huge model file). 

Notably, I converge to a far smaller loss when using this feature, but of course the resulting model is probably too huge to be valid/usable.

**Comment by matanster** [comment]
Thanks a lot, that makes sense, which is more than I can say about my trying out this feature combination. And in line with VW's philosophy of letting users figure which options don't make sense to combine. I guess I'll never know why this got me far better training loss than without turning this flag on :-)

**Comment by JohnLangford** [comment]
I _could_ imagine just storing the random seed for the random weights, but that's a "patches welcome" type of thing.

It's odd that this gave significantly better performance.  Could that be just a random luck thing?  Anyways, I'm closing this one for now.
